.github/workflows/build-api-ref.yml

```diff
@@ -0,0 +1,41 @@
+name: Generate API Reference Documentation
+
+on:
+  push:
```

Comment:
Probably should be added filtering by main branch also, because e.g. sometimes I could make some changes in other branches which could led to trigger this workflow.